packages/medusa-file-minio/src/services/minio.ts

```diff
@@ -2,10 +2,29 @@ import stream from "stream"
 import aws from "aws-sdk"
 import { parse } from "path"
 import fs from "fs"
-import { AbstractFileService } from "@medusajs/medusa"
+import {
+  AbstractFileService,
+  DeleteFileType,
+  FileServiceUploadResult,
+  GetUploadedFileType,
+  IFileService,
+  UploadStreamDescriptorType,
+} from "@medusajs/medusa"
 import { MedusaError } from "medusa-core-utils"
+import { ClientConfiguration, PutObjectRequest } from "aws-sdk/clients/s3"
+
+class MinioService extends AbstractFileService implements IFileService {
+  private bucket_: string
```

Comment:
todo: Make them protected so that if anybody want to add custom methods by extending this service they will have access to those props